sql/rename.go

```diff
@@ -209,6 +213,8 @@ func (p *planner) RenameIndex(n *parser.RenameIndex) (planNode, error) {
 	if err := p.txn.Put(descKey, tableDesc); err != nil {
 		return nil, err
 	}
+	// Mark transaction as operating on the system DB.
+	p.txn.SetSystemDBTrigger()
```

Comment:
As other's have mentioned, requiring these calls to be sprinkled whenever the system DB span is modified seems fragile. Gossiping of the old-style configs is done by `maybeGossipConfigs`. Why can't we rely on something similar here?